src/executor/infer_graph_attr_pass.cc

```diff
@@ -0,0 +1,306 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file infer_graph_attr_pass.cc
+ * \brief infer graph shape, dtype, and storage type
+ */
+
+#include <mxnet/op_attr_types.h>
+#include "./exec_pass.h"
+
+namespace mxnet {
+namespace exec {
+
+template<typename AttrType, typename IsNone, typename FDefault>
+nnvm::Graph InferAttr(nnvm::Graph &&ret,
```

Comment:
We probably need a separate interface for InferStorage which takes an attr(storage_type) and aux attribute (ctx)